docs/dev/version-migration/migration-guide.md

```diff
@@ -0,0 +1,22 @@
+# `partiql-lang-kotlin` Version Migration Guide
+
+(Note: this folder's documents will likely be converted to a set of wikis on https://github.com/partiql/partiql-lang-kotlin/wiki
+with each major version bump on a separate wiki page.
+TBD if these documents will remain as a markdown file on GitHub)
```

Comment:
+1 to wiki, but can be a separate task so we can get this merged

Reply:
Sounds good. Once this PR is approved, I'll have a separate task to copy the guides to wikis.